ops/services/app_service/main.tf

```diff
@@ -66,3 +106,24 @@ resource "azurerm_key_vault_access_policy" "app_secret_access" {
     "list",
   ]
 }
+
+resource "azurerm_key_vault_access_policy" "staging_slot_secret_access" {
+  key_vault_id = var.key_vault_id
+  object_id    = azurerm_app_service_slot.staging.identity[0].principal_id
+  tenant_id    = var.tenant_id
+
+  key_permissions = [
+    "get",
+    "list",
+  ]
+  secret_permissions = [
+    "get",
+    "list",
+  ]
+}
```

Comment:
I see you didn't end up using the hack I had put in ReportStream's resources (the one using count). We're going to need it because on the deploy to other environments, the service principal will not be created until the slot is created the first time.

Reply:
D'oh. Yeah, I forgot I hadn't done that. Though weirdly it does seem to work when I spin it up all at once, just not when I add the identity block and permissions block as updates to an existing slot (which is why forgetting it was possible, at least for a day or two).

Reply:
I did some testing myself recently, and that does seem to be the case. If the resource does not exist, Terraform creates everything fine. If the resource does exist, Terraform requires two deployments for the service principal.

Also, if the count is added and the resource does not exist, Terraform complains on the `plan` but not the `apply`. I'm thinking this a bug in the Terraform provider.